CMakeLists.txt

```diff
@@ -46,6 +46,15 @@ if(USE_CUDA AND NOT USE_OLDCMAKECUDA)
         (${CMAKE_VERSION} VERSION_GREATER "3.9.0") OR (${CMAKE_VERSION} VERSION_EQUAL "3.9.0")
       )
     )
+    
+    set(__cuda_toolset "auto" "7.5" "8.0" "9.0")
```

Comment:
this can’t go here above all of the config options

Reply:
You mean you want to automatically detect the installed version of the CUDA?

Reply:
it already is auto detecting and offering an override option.
what doesn’t it work on your machine but it works elsewhere?
also there should be very little above the main config options

Reply:
or add 9.1?

Reply:
My build machine install multiple versions cuda,It needs to be compiled with different versions of CUDA.like cu80 cu90 cu91 etc.and auto detecting always use latest version cuda.

Reply:
right i mean without any of these changes. the problem is it just picks the latest one?
the current implementation allows for setting the cuda toolset version, doesn’t it?